integration-tests/src/test/java/oracle/kubernetes/operator/ITStickySession.java

```diff
@@ -0,0 +1,307 @@
+// Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at
+// http://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import oracle.kubernetes.operator.utils.Domain;
+import oracle.kubernetes.operator.utils.ExecResult;
+import oracle.kubernetes.operator.utils.Operator;
+import oracle.kubernetes.operator.utils.TestUtils;
+import org.junit.AfterClass;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+/**
+ * Simple JUnit test file used for testing Operator.
+ *
+ * <p>This test is used for testing the affinity between a web client and a Weblogic server for the
+ * duration of a HTTP session created by Voyager load balancer.
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class ITStickySession extends BaseTest {
+  private static final String testAppName = "stickysessionapp";
+  private static final String testAppPath = testAppName + "/StickySessionCounterServlet";
+  private static final String scriptName = "buildDeployAppInPod.sh";
+
+  private static Map<String, String> httpAttrMap;
+
+  private static String httpHeaderFile;
+
+  private static Operator operator;
+  private static Domain domain;
+
+  /**
+   * This method gets called only once before any of the test methods are executed. It does the
+   * initialization of the integration test properties defined in OperatorIT.properties and setting
+   * the resultRoot, pvRoot and projectRoot attributes. It creates an operator, a Weblogic domain
+   * and deploy a web application with persistent-store-type configured to replicated_if_clustered.
+   *
+   * @throws Exception
+   */
+  @BeforeClass
+  public static void staticPrepare() throws Exception {
+    if (!QUICKTEST) {
+      // initialize test properties and create the directories
+      initialize(APP_PROPS_FILE);
+
+      // Create operator1
+      if (operator == null) {
+        logger.info("Creating Operator & waiting for the script to complete execution");
+        operator = TestUtils.createOperator(OPERATOR1_YAML);
+      }
+
+      // create domain
+      if (domain == null) {
+        logger.info("Creating WLS Domain & waiting for the script to complete execution");
+        // Treafik doesn't work due to the bug 28050300. Use Voyager instead
+        System.setProperty("LB_TYPE", "VOYAGER");
```

Comment:
Is there a reason why you are using system property instead of setting loadBalancer in domain map?

Reply:
Please check the changes I made in Domain.java, line 1297, we can only switch to use Voyager if a env variable is true. I discussed this with Maggie, we concluded that I add a System property in order to use Voyager

Reply:
You can use voyager by setting/changing loadBalancer and loadBalancerWebPort attributes also in the domain map at runtime. No need to add a system property.
```
      Map<String, Object> domainMap = TestUtils.loadYaml(DOMAINONPV_WLST_YAML);
      domainMap.put("loadBalancer", "VOYAGER");
      domain = TestUtils.createDomain(domainMap);
```

Reply:
oh. ok let me give it a try